### PR TITLE
Custom DomainType and TrustedPeers (Alan)

### PR DESCRIPTION
**Pull request by guym-blox**
### Description

this fix https://github.com/ssvlabs/ssv/issues/1565 (#1493) provide custom domain and trusted peers

### Custom domain and forking

We have decided custom domain isn't relevant after alan fork so this change means custom domain overrides genesis domain and after alan will fork everyone will be on the same domain.

### CODE REVIEW

**Review by y0sher — APPROVED** [review]
lgtm

File: cli/operator/node.go [hunk 1]
```diff
@@ -594,6 +596,21 @@ func setupSSVNetwork(logger *zap.Logger) (networkconfig.NetworkConfig, error) {
...
+		}
+		byts, err := hex.DecodeString(cfg.SSVOptions.CustomDomainType[2:])
+		if err != nil {
+			return networkconfig.NetworkConfig{}, errors.Wrap(err, "failed to decode custom domain type")
+		}
+		if len(byts) != 4 {
+			return networkconfig.NetworkConfig{}, errors.New("custom domain type must be 4 bytes")
+		}
+		networkConfig.GenesisDomainType = spectypes.DomainType(byts)
+		logger.Info("running with custom domain type", fields.Domain(networkConfig.GenesisDomainType))
```
Comment (moshe-blox) [inline cli/operator/node.go:611]: should probably add something like:
```suggestion
if networkConfig.PastAlanFork() {
    logger.Info("custom domain type is ineffective now after Alan fork", fields.Domain(networkConfig.GenesisDomainType))
} else {
    logger.Info("running with custom domain type", fields.Domain(networkConfig.GenesisDomainType))
}
```

File: network/p2p/p2p.go [hunk 2]
```diff
@@ -172,6 +201,43 @@ func (n *p2pNetwork) Close() error {
...
+		return nil, err
+	}
+	maStrs := make([]string, len(ma))
+	for i, ima := range ma {
+		maStrs[i] = ima.String()
+	}
+	logger.Info("starting p2p",
+		zap.String("my_address", strings.Join(maStrs, ",")),
+		zap.Int("trusted_peers", len(n.trustedPeers)),
+	)
```
Comment (moshe-blox) [inline network/p2p/p2p.go:237]: this part is unrelated to getConnector, could be extracted to logStartup or just moved back into Start()

**Review by moshe-blox — CHANGES_REQUESTED** [review]

**Comment by moshe-blox** [comment]
well done, just 2 comments

**Review by moshe-blox — APPROVED** [review]

**Pull request merged**